_rules/SC4-1-2-iframe-unique-name.md

```diff
@@ -0,0 +1,144 @@
+
+- [http://www.w3.org/TR/WCAG20-TECHS/H64.html](http://www.w3.org/TR/WCAG20-TECHS/H64.html)
+- [https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat-rsv](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#ensure-compat-rsv)
+
+## Test cases
+
+### Passed
+
+#### Pass example 1
+
+Usage of `title` attribute to describe the `iframe` content, and there is only one iframe with in document context.
+
+```html
+<iframe title="List of Contributors" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+```
+
+#### Pass example 2
+
+Multiple `iframe` elements in the document having different `title` descriptions as accessible name.
+
+```html
+<iframe title="List of Contributors to Repository 1" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors to Repository 2" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Pass example 3
+
+Multiple `iframe` elements in the document having different `aria-label` descriptions as accessible name.
+
+```html
+<iframe aria-label="List of Contributors to Repository 1" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors to Repository 2" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Pass example 4
+
+Multiple `iframe` elements in the document having different `aria-labelledby` descriptions as accessible name.
+
+```html
+<div id="desc-for-title">List of Contributors</div>
+<iframe aria-labelledby="desc-for-title" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<div id="desc-for-title1">List of Reviewers</div>
+<iframe aria-labelledby="desc-for-title1" role="document" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+### Failed
+
+#### Fail example 1
+
+Multiple frames having the same `title` within a given document context.
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Fail example 2
+
+Multiple frames having the same `aria-label` within a given document context.
+
+```html
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+#### Fail example 3
+
+Multiple frames having the same name given by `title` and `aria-label` within a given document context
+
+```html
+<iframe title="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc1.html">
+</iframe>
+<iframe aria-label="List of Contributors" src="../test-assets/SC4-1-2-iframe-unique-name-doc2.html">
+</iframe>
+```
+
+### Inapplicable
```

Comment:
Should test frames without an accessible name.

Reply:
Sorted